### PR TITLE
feat(functions): use configurations, update to functions

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- change to composition functions
- enable renovate
- use configurations like configuration-gcp-network/gke/database

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
--- PASS: kuttl (3042.59s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (3042.09s)
PASS
10:48:15 [ OK ] running automated tests
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,13 +11,15 @@ PLATFORMS ?= linux_amd64
 # ====================================================================================
 # Setup Kubernetes tools
 
-UP_VERSION = v0.14.0
+UP_VERSION = v0.21.0
 UP_CHANNEL = stable
-UPTEST_VERSION = v0.2.1
+UPTEST_VERSION = v0.9.0
```
Comment (ytsarev) [inline Makefile:16]: (nit) we can use latest right away
```suggestion
UPTEST_VERSION = v0.10.0
```

File: README.md
Comment (ytsarev) [inline README.md:1]: It's not the part of the diff, so I can't suggest on the specific line. 

Please switch provider link from monolith https://marketplace.upbound.io/providers/upbound/provider-gcp to https://marketplace.upbound.io/providers/upbound/provider-family-gcp/v0.41.0

File: examples/app-claim.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: default
+  labels:
+    platform.upbound.io/deletion-ordering: enabled
+spec:
+  compositeDeletePolicy: Foreground
+  parameters:
+    providerConfigName: platform-ref-gcp
+    passwordSecretRef:
+      namespace: default
+      name: platform-ref-gcp-db-conn-mariadb
```
Comment (ytsarev) [inline examples/app-claim.yaml:14]: it's not working until we add mariadb support to https://github.com/upbound/configuration-gcp-database, correct? Just to double check that it is a placeholder

**Review by ytsarev — COMMENTED** [review]
It looks great, left some minor comments

File: examples/app-claim.yaml (hunk 2 above)
Comment (haarchri) [reply examples/app-claim.yaml:14]: Yes

File: Makefile (hunk 1 above)
Comment (haarchri) [reply Makefile:16]: Will merge all renovate after this PR

File: README.md
Comment (haarchri) [reply README.md:1]: good catch

Comment (haarchri) [reply README.md:1]: changed - will not run uptest again because of readme change okay ? ;)

**Review by ytsarev — APPROVED** [review]
thanks for the fixes

**Pull request merged**